### DISCUSSION TITLE
Fixture : scope "session" not working as expected

### DISCUSSION THREAD

**Discussion by harmin-parra** [pytest-dev/pytest#8526] [Q&A]
I have a function with the following decorator
**@pytest.fixture(scope = "session")**

The setup fixture is used in 2 different pytest **.py** files

Execute all tests from the command line

Actual result
The setup fixture is executed twice

Expected result
The setup fixture is executed once

Pytest version : 6.2.2

**Comment by RonnyPfannschmidt** [comment]
Without code we have no idea what you do

Do you use a contest or import?

**Comment by harmin-parra** [comment]
I have 3 files :

file1.py
```
@pytest.fixture(scope = "session")
def setup() :
	some code
```

file2.py
```
import file1
def test_1(setup) :
	some code
```

file3.py
```
import file1
def test_2(setup) :
	some code
```

Execute with **pytest file2 file3**

**Reply by The-Compiler** [reply]
Please share an example which actually runs and where you can verify it reproduces your issue. With your example, even after:

- Replacing "some code" with `pass`
- Adding the missing `pytest` import
- Changing the command to `pytest file2.py file3.py`

This won't run because the `setup` fixture isn't available in `file2.py` or `file3.py`. At this point, this is too far away from the actual situation to say anything.

**Reply by harmin-parra** [reply]
Create a folder (Ex: "**github**")

In "**github**" folder create the following files :
_ init _.py
file2.py
file3.py

Contents of each file:

_ init _.py
```
import pytest

@pytest.fixture(scope = "session")
def setup() :
    print("I am executing")
```
file2.py
```
from github import *

def test_1(setup) :
    pass
```
file3.py
```
from github import *

def test_2(setup) :
    pass
```
Execute the command **pytest -rA github/file2.py github/file3.py**

Output:

```
============================= test session starts ==============================
platform linux -- Python 3.8.5, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
rootdir: /home/harmin/workspace/python/dev
plugins: metadata-1.11.0, html-3.1.1
collected 2 items                                                              

github/file2.py .                                                        [ 50%]
github/file3.py .                                                        [100%]
Summary: 2 passed

==================================== PASSES ====================================
____________________________________ test_1 ____________________________________
---------------------------- Captured stdout setup -----------------------------
I am executing
____________________________________ test_2 ____________________________________
---------------------------- Captured stdout setup -----------------------------
I am executing
=========================== short test summary info ============================
PASSED github/file2.py::test_1
PASSED github/file3.py::test_2
```
**There you can see that the setup fixture is executed twice instead of once.**

**Reply by The-Compiler** [reply]
If you do `from github import *` (i.e. import the fixture), it exists twice. Don't do that, use a [conftest.py](https://docs.pytest.org/en/stable/fixture.html#conftest-py-sharing-fixtures-across-multiple-files) instead.
